### PR TITLE
Don't skip building arrow if it hasn't successfully been built.

### PR DESCRIPTION
**Pull request by robertnishihara**
cc @richardliaw

### CODE REVIEW

File: thirdparty/scripts/build_arrow.sh [hunk 1]
```diff
@@ -35,9 +35,12 @@ else
 fi
 
 # Download and compile arrow if it isn't already present.
-if [[ ! -d $TP_DIR/pkg/arrow ]]; then
+if [[ ! -d $TP_DIR/../python/ray/pyarrow_files/pyarrow ]]; then
     echo "building arrow"
-    git clone https://github.com/apache/arrow.git "$TP_DIR/build/arrow"
+
+    if [[ ! -d $TP_DIR/build/arrow ]]; then
+      git clone https://github.com/apache/arrow.git "$TP_DIR/build/arrow"
```
Comment (richardliaw) [inline thirdparty/scripts/build_arrow.sh]: this doesn't fail the build right?

Comment (robertnishihara) [reply thirdparty/scripts/build_arrow.sh]: no it doesn't, the build should fail only if `make VERBOSE=1 -j$PARALLEL` below fails, I think

Comment (richardliaw) [reply thirdparty/scripts/build_arrow.sh]: wait is it supposed to? what is the name of the PR referring to?

Comment (robertnishihara) [reply thirdparty/scripts/build_arrow.sh]: You're right, the name is wrong, I'll fix it.

**Review by pcmoritz — APPROVED** [review]

**Pull request merged**